### DISCUSSION TITLE
Mapping ensembles to rdd or bag

### DISCUSSION THREAD

**Discussion by pavlis** [mspass-team/mspass#123] [Design & Development]
I think we very much need a way to map any ensemble into an rdd or bag (I think that is the dask equivalent).   The reason is I can see a need for strings of ensemble-oriented processes that could exploit that.   The case in point is the raw data handling I'm current deep into debugging.   Seems to me this kind of pseudocode would be helpful when handling large ensembles of data like usarray gathers:
```
1) read ensemble data
2) convert ensemble to rdd
3) parallel filter
4) parallel window
5) parallel bundle
6) parallel write
```
Step 2 is required to do that, but I'm too much of a rookie to write it.  

Note step 1 might be considered a bottleneck, but if my test data means anything it takes only a few seconds to read the raw seed ensemble data but the rest of the steps are taking a bit over one minute per ensemble.   Thus, throwing a handful of processors at the job could improve performance a lot.

**Comment by wangyinz** [comment]
Step 2 can be done with 
```
bbg = dask.bag.from_sequence(ens.member)
```
or
```
rdd = spark_context.parallelize(ens.member)
```
Do we really need a wrapper for that?

**Comment by pavlis** [comment]
No we don't need a wrapper if it is that easy, BUT we DEFINITELY need to have that in our documentation of the parallel api.

**Comment by wangyinz** [comment]
Well, note that this will drop the ensemble's metadata, so we should also document that and make sure the users call the `sync_metadata` method first if they want to retain that.
